konrad/core.py

```diff
@@ -228,8 +238,28 @@ def is_converged(self):
         Returns:
             bool: ``True`` if converged, else ``False``.
         """
-        # TODO: Implement proper convergence criterion (e.g. include TOA).
-        return np.all(np.abs(self.deltaT) < self.delta)
+        
+        self.newDN = self.newN - self.oldN
```

Comment:
Here, you can still access the model components. So it is possible to do things like
`self.radiation["toa"][-1] - self.oldN`
Therefore there is no need to store the current (`new`) value of each variable.